modules/Bio/EnsEMBL/Hive/DBSQL/AnalysisJobAdaptor.pm

```diff
@@ -587,6 +587,57 @@ sub reset_or_grab_job_by_dbID {
     return $job;
 }
 
+=head2 reset_job_by_input_id_and_sync
+
+  Arg [1]: string $input_id pattern
+  Arg [2]: array $analsyes_object_array
+  Example:
+    my $job = $self->{'dba'}->get_AnalysisJobAdaptor->reset_job_by_input_id_and_sync($input_id_pattern, $analsyes_object_array);
+  Description:
+    Reset jobs for the specified $input_id pattern and $analsyis_object_array. $input_id can be a wildcard argument.
+  Returntype : none
+  Exceptions :
+  Caller     : beekeeper.pl
+
+=cut
+
+sub reset_job_by_input_id_and_sync {
+    my ($self, $input_id_pattern, $analyses_list) = @_;
+
+    #Convert MySQL wildcard pattern to Perl wildcard pattern
+    $input_id_pattern = return_qr_pattern($input_id_pattern);
+
+    my $analysis;
+    my $jobs = $self->fetch_all_by_analysis_id_status($analyses_list);
+    foreach my $job (@$jobs) {
+         my $input_id = $job->input_id;
+         if ($input_id =~ /$input_id_pattern/) {
+                my $job_status = $job->status();
+                my $job_info = $job->toString;
+                if($job_status =~ /$ALL_STATUSES_OF_RUNNING_JOBS|CLAIMED|SEMAPHORED/) {
+                   warn "$job_info is $job_status, cannot reset - skipping to next job";
+                   next;
+                }
+
+                #The job at this stage will be DONE, PASSED_ON, FAILED or READY
+                if ($job_status ne 'READY') {
```

Comment:
The negative test here is a bit strange. The previous `if` follows the paradigm "detect and process the special cases early". READY jobs are the same: print a warning and move on. I would like the same approach: if the job is READY, `warn` and `next`.